Add docs for new iteration of ent operator import feature (#29406)
hashicorp/vault: website/content/docs/import/index.mdx

```hcl
source_gcp {
  name             = "my-gcp-source-1"
  credentials_file = "/path/to/service-account-key.json"
}
```

- `name` `(string: <required>)` - A unique name for the source block that can be
  referenced in subsequent mapping blocks.

- `credentials_file` `(string: <required>)` - Path to a credential file with
  read permissions for the target system.

Depending on the source system, additional information may be required. Refer to
the connection documentation for your source system to determine the full set of
required fields for that system type.

## Mappings

Mappings glue the source and destination together and filter the migrated data,
to determine what is imported and what is ignored.

### HCL syntax

> Mapping blocks require a name, a source name, destination name, and any corresponding

Reviewer: ```suggestion
Mapping blocks require a name, a source name, a destination name, and any corresponding
```

- `destination` `(string: <required>)` - The name of a previously defined
  destination block **to** which the data should be written.

> - `filter` `(string: "")` - Optional string containing a https://github.com/hashicorp/go-bexpr
>   expression to limit which secrets from the source will be imported.

Reviewer 2: ```suggestion
- `filter` `(string: "")` - Optional string containing a (`bexpr` style boolean expression)[https://github.com/hashicorp/go-bexpr]
 that limits which secrets Vault imports from the source.
```

#### Filters

> A filter describes conditions that a source secret must meet in order to be imported.
> These conditions are applied to the secret fields, which are:

Reviewer 2: ```suggestion
Filters describes conditions for secret fields that a source secret must meet to be imported. You can filter against the following secret fields:
```
Style correction: write in the active voice

- `Secret.Name` - The name of the source secret
- `Secret.Tags` - A map of key-value user-defined metadata associated with the source secret

> Simple binary conditions can be applied to input values:

Reviewer 2: ```suggestion
You can also apply simple binary conditions to input values:
```
Style correction: write in the active voice

- `Secret.Name != "foo"`
- `Secret.Tags.Team matches "my-dept.*"`
- `Secret.Tags contains "my-required-tag-name"`

These can be combined using `and`, `or`, and `not`:

- `(Secret.Name == "foo" or Secret.Tags.Team == "my-team") and Secret.Tags.Category = "my-cat"`

#### Transformers

Transform stanzas come in two forms: `exact` and `regexp`.

> An exact transform allows renaming a secret during import, so that the `from` secret name
> is imported into Vault as a secret named `to`.  In the example below, a source secret named
> `foo` will be imported as `foosball`.

Reviewer 2: ```suggestion
`exact` transforms rename secrets during import so the `from` secret name becomes the `to` secret name. In the following example , takes a source secret named
`foo` and transforms it to `foosball` during import.
```
Style correction: write in the active voice

Author: I think "In the following example , takes a source secret named `foo` and transforms it to `foosball` during import" doesn't sound quite right, there seems to be a missing subject.  I'll try it with "it" as the subject.

Author: Attempted rewrite in 6f7c69cb1a43583263b6691ee1631c8e03ef9575

```hcl
  transform "exact" {
    from = "foo"
    to = "foosball"
  }
```

> A regexp transform allows renaming a secret during import using [Go regular expression syntax](https://github.com/google/re2/wiki/Syntax).
> Parentheses designate capture groups that are used to define the name that secrets matching the regular
> expression will be imported as.  In the example below, any secret in the source whose name
> starts with "foo" will have the leading "foo" prefix replaced with "bar".  Named capture groups
> using the syntax `(?P<name>re)` are also supported, in which case the replacement would be addressed
> as `$name` instead of a numeric reference like `$1`.
> 
> ```hcl
>   transform "regexp" {
>     from = "foo(.*)"
>     to = "bar$1"
>   }
>   transform "regexp" {
>     from = "foo(?P<suffix>.*)"
>     to = "bar$suffix"
>   }
> ```

Reviewer 2: ````suggestion
`regexp` transforms rename secrets during import using
[Go regular expression syntax](https://github.com/google/re2/wiki/Syntax). For
example, the transform below imports any secret in the source whose name starts
with `foo` and replaces the `foo` prefix with `bar`.  

```hcl
  transform "regexp" {
    from = "foo(.*)"
    to = "bar$1"
  }
  transform "regexp" {
    from = "foo(?P<suffix>.*)"
    to = "bar$suffix"
  }
```

The `from` value supports parentheses to bookend capture groups and named
capture groups using the syntax `(?P<name>re)`. When you use named capture
groups, you can reference the named group in the `to` value. For example, 
`$name` instead of `$1`.
````
Style correction: write in active voice, provide the example before discussing the example text in detail
